portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -15,18 +15,30 @@
 package com.google.sps.servlets;
 
 import java.io.IOException;
+import java.util.ArrayList;
+
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+import com.google.gson.Gson;
+
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private ArrayList<String> commentsString = new ArrayList<String>();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // hard coded values for array
+    commentsString.add("hello ");
+    commentsString.add("this is ");
+    commentsString.add("a random comment ");
```

Comment:
You could make this an [ImmutableList](https://guava.dev/releases/23.0/api/docs/com/google/common/collect/ImmutableList.html) instead, and initialise it like they do [here](https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names).

Generally when we have a variable that doesn't change (e.g. hardcoded) it is better to make it immutable / constant, because then the next person who reads the code _knows_ that it is a constant (i.e. they won't have to worry about it changing). [There are some other reasons for using immutable objects here as well.](https://g3doc.corp.google.com/third_party/java_src/google_common/current/java/com/google/common/collect/g3doc/immutable-collections.md?cl=head)

Reply:
Yep, will change my ArrayLists to Immutable Lists

Reply:
Oh wait, in the final version, we do change the list. The list is only hardcoded here as per the walkthrough.